pandas/core/indexes/datetimelike.py

```diff
@@ -754,44 +593,6 @@ def isin(self, values):
 
         return algorithms.isin(self.asi8, values.asi8)
 
-    def shift(self, n, freq=None):
```

Comment:
Should we keep a stub calling the super method here as well? (as you did with the arithmetic methods or with `_generate_range`) 
I know it is not strictly needed here, but I have the feeling that will make it more clear that then in the future we need to update this from calling super class method to calling the composited values method (although I suppose we have tests for it that will also be a reminder to do it)

Reply:
For `_generate_range` and a few others it is needed because the Index method needs to set `result.name`.

There are a handful of other things things in the same situation as shift (e.g. several properties that are overriden by cache_readonlys) that will need to be edited if/when the change from subclassing to composition is made.  I'll make a note not to forget `shift` when the time comes.